src/BudgetManager.Shared/BudgetManager.Shared.DataAccess.MongoDB/BaseImplementation/BaseRepository.cs

```diff
@@ -0,0 +1,76 @@
+﻿using System.Linq.Expressions;
+using BudgetManager.Shared.DataAccess.MongoDB.DatabaseSettings;
+using BudgetManager.Shared.Models.MongoDB.Models.Interfaces;
+using MongoDB.Driver;
+using MongoDbGenericRepository.Attributes;
+
+namespace BudgetManager.Shared.DataAccess.MongoDB.BaseImplementation
+{
+    public class BaseRepository<TDocument> : IBaseRepository<TDocument>
+        where TDocument : IModelBase
+    {
+        private readonly IMongoCollection<TDocument> _collection;
+        public BaseRepository(IMongoDbSettings settings)
+        {
+            var database = new MongoClient(settings.ConnectionString).GetDatabase(settings.DatabaseName);
+            _collection = database.GetCollection<TDocument>(GetCollectionName(typeof(TDocument)));
```

Comment:
Need to add following string here because our Id is storing like Binary not UUID
```suggestion
            BsonSerializer.RegisterSerializer(new GuidSerializer(GuidRepresentation.Standard));
            var database = new MongoClient(settings.ConnectionString).GetDatabase(settings.DatabaseName);
            _collection = database.GetCollection<TDocument>(GetCollectionName(typeof(TDocument)));
```